wasi_snapshot_preview1/fs.go

```diff
@@ -411,7 +411,12 @@ var fdRead = wasm.NewGoFunc(
 			if errors.Is(err, io.EOF) {
 				break
 			} else if err != nil {
+				if n != 0 { // Let callers process the n > 0 bytes returned before considering the error err.
+					break
```

Comment:
doesn't this swallow the error? how can they consider the err if there's none returned?

Reply:
TLDR: you handle the error on your next call to `readv` (`fread`, etc).

If you assume the error is stable (e.g. an external disk was removed), when you `readv` again you'll get the same error.

OTOH if the error is transient (which is unlikely for real filesystems), and you call `readv` again, you'll have lost data and will have no way to read it back again.

So yes, this may swallow errors in 2 situations:
1. if `n` is enough for the caller, and it doesn't `readv` again.
2. if `err` is transient, and isn't returned on a future call to `readv`.